src/client/datascience/editor-integration/codeLensFactory.ts

```diff
@@ -184,7 +184,7 @@ export class CodeLensFactory implements ICodeLensFactory {
             };
             this.notebookData.set(e.cell.notebook.uri.toString(), data);
         }
-        if (data) {
+        if (data && metadata.interactive !== undefined) {
```

Comment:
Please can we use `getInteractiveCellMetadata` function.
Else tomorrow if we change the types/names its impossible to find out the usages.
I.e. use types as opposed to `any`

Reply:
E.g. when I had to change `executionId` to `id`, it wasn't easy to find all the usages, and easy to miss usages and break the code.